pkg/cloud/services/compute/instance.go

```diff
@@ -653,9 +653,28 @@ func (s *Service) GetManagementPort(instanceStatus *InstanceStatus) (*ports.Port
 	return &allPorts[0], nil
 }
 
-func (s *Service) DeleteInstance(eventObject runtime.Object, instance *InstanceIdentifier) error {
+func (s *Service) garbageCollectErrorInstancesPort(eventObject runtime.Object, instance *InstanceStatus) error {
+	pager, err := ports.List(s.networkClient, ports.ListOpts{
+		Name: instance.Name(),
+	}).AllPages()
+	if err != nil {
+		return err
+	}
+	portList, err := ports.ExtractPorts(pager)
+	if err != nil {
+		return err
+	}
+	if len(portList) < 1 {
+		return nil
+	}
+
+	return s.deletePort(eventObject, portList[0].ID)
```

Comment:
Even if we only support a single port per instance I would suggest to loop over the `portList` and delete all ports. It does not hurt and it feels more stable.
WDYT?
cc @jichenjc

Reply:
Also we don't have to check for `if len(portList) < 1` in this case

Reply:
Thanks @tobiasgiese, I agree it and modify code to loop through portList and delete each port.

Reply:
I fixed it. What about it ? @tobiasgiese

Reply:
Perfect, thanks :)

Please squash your commits. Then I will give my lgtm

Reply:
Ok, I squashed.